### PR TITLE
Fix duration calculation

### PR DESCRIPTION
**Pull request by wxiaoguang**
Fix duration calculation

The simplest fix for https://github.com/github/relative-time-element/issues/250#issuecomment-2509670103 , not ideal, just a quick fix.

### CODE REVIEW

**Comment by silverwind** [comment]
Maybe a test can be added here: 

https://github.com/github/relative-time-element/blob/e914da24dad0980cbfcb5dae6d56cf2c97659340/test/duration.ts#L231

**Comment by wxiaoguang** [comment]
> Maybe a test can be added here:
> 
> https://github.com/github/relative-time-element/blob/e914da24dad0980cbfcb5dae6d56cf2c97659340/test/duration.ts#L231

If the project members have interests in this PR, I could add later.

**Comment by francinelucca** [comment]
Thanks so much for diving into this @wxiaoguang! Looks like this proposed solution is causing one test to fail so we're unable to merge it at this time.
If you'd like to take a stab at fixing the tests you're welcome to do so, if not someone from the team will pick this up when we have the availability to do so.

You should be able to run the tests locally using `npm run test`

**Comment by wxiaoguang** [comment]
> Thanks so much for diving into this @wxiaoguang! Looks like this proposed solution is causing one test to fail so we're unable to merge it at this time. If you'd like to take a stab at fixing the tests you're welcome to do so, if not someone from the team will pick this up when we have the availability to do so.
> 
> You should be able to run the tests locally using `npm run test`

But that failure seems not related. I have noticed that even on main branch, it still fails (that's another reason why I didn't touch the test code, I just would like to see whether it fails in your CI tasks).

````
 ❌ relative-time > [tense=future] > micro formats years
      AssertionError: expected '11y' to equal '10y'
      + expected - actual
      
      -11y
      +10y
      
      at n4.<anonymous> (test/relative-time.js:609:13)
````

**Comment by wxiaoguang** [comment]
The `11y vs 10y`  failure is caused by another bug in `elapsedTime`, `const month = Math.floor(day / 30)` is not right when the period is long enough.

File: src/duration.ts [hunk 1]
```diff
@@ -182,6 +182,9 @@ export function roundToSingleUnit(duration: Duration, {relativeTo = Date.now()}:
     } else if (monthsDiff < 11) {
       months = monthsDiff
       years = 0
+    } else if (monthsDiff === 11) {
+      months = 0
+      years += 1 // the old behavior: "11 months" is rounded to "1 year"
```
Comment (francinelucca) [inline src/duration.ts:187]: so in this case if the two dates were January and December, we'd round up to a year? wouldn't we want it to be 11 months?

**Review by francinelucca — COMMENTED** [review]
Confirmed CI failure is present in prod, in that case I think we can move forward with this. @wxiaoguang Just have a logic question and adding tests I think is a great idea :)

File: src/duration.ts (hunk 1 above)
Comment (wxiaoguang) [reply src/duration.ts:187]: > so in this case if the two dates were January and December, we'd round up to a year? wouldn't we want it to be 11 months?

The old code didn't do so due to ` if (monthsDiff < 11) { } else { year }` above

I have tried to make it support "11 months", it breaks a lot of tests.

**Comment by wxiaoguang** [comment]
A test is added in fc858c3c34e038476008be60004b33ddfd368707 and added comments to clarify the problem.

**Comment by wxiaoguang** [comment]
And made one more commit: if we'd like to support "11 months": 75800544934caeecdcd841dbe583ceb410a502e2 corrected the behavior and fixed the tests.

Feel free to choose the behavior you like.

**Comment by wxiaoguang** [comment]
"Fixed" the "10y vs 11y" problem in da833cf59b93e06d91d314001b518d78219ec809, now the CI should be able to pass

**Review by viceice — APPROVED** [review]
🚀 ❤️

**Review by keithamus — APPROVED** [review]
This LGTM

**Review by francinelucca — APPROVED** [review]

**Comment by francinelucca** [comment]
Merged! Thanks so much for doing this @wxiaoguang

**Pull request merged**